### PR TITLE
MRG: Fixes for TRIUX and MEG ordering

### LINKED ISSUE #3221: maxwell_filter not working with Triux files - channel order issue
I've been trying to Maxwell filter Triux fif files and have been encountering issues related to the channel ordering. Because Triux files put EOG etc data before MEG, `_update_sensor_geometry` fails on line 1392. 
 
```
    grad_picks = pick_types(info, meg='grad', exclude=[])
    mag_picks = pick_types(info, meg='mag', exclude=[])
    grad_imbalances = np.array([cal_chs[ii]['calib_coeff']
                                for ii in grad_picks]).T
```

The channel numbers returned by grad-picks do not align with the size of `cal_chs` which lead to an out of range index error. 
Fixing this problem with a dirty hack did not suffice, as further matrix size mismatches appeared in `_sss_basis_point` at line 1427:
 `       S_add *= imb[:, np.newaxis]`

I am not sure how to properly fix this.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Fixes some ugly indexing bugs, and tests against TRIUX empty-room data provided by @Mountainmums.

Ready for review/merge from my end. @wronk do you have time to take a quick look? You'll recognize some of the changes from #2818, which I'll rebase after this is in.

Closes #3221.

### CODE REVIEW

**Comment by wronk** [comment]
Will do

File: mne/preprocessing/_fine_cal.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    """
+    _check_fname(fname, overwrite=True)
+    check_fname(fname, 'cal', ('.dat',))
+
+    with open(fname, 'wb') as cal_file:
+        for ci, chan in enumerate(calibration['ch_names']):
+            # Write string containing 1) channel, 2) loc info, 3) calib info
+            cal_line = np.concatenate([calibration['locs'][ci],
+                                       calibration['imb_cals'][ci]]).round(6)
+            cal_str = str(chan) + ' ' + ' '.join(map(lambda x: "%.6f" % x,
```
Comment (wronk) [inline mne/preprocessing/_fine_cal.py]: For posterity, maybe add a comment above to indicate why 6 decimal places.

`# Use 6 decimal places to conform to Elekta's original implementation.`

File: mne/preprocessing/maxwell.py [hunk 2]
```diff
@@ -341,29 +343,32 @@ def maxwell_filter(raw, origin='auto', int_order=8, ext_order=3,
...
-        max_st.update(job=10, subspcorr=st_correlation, buflen=st_duration)
+        max_st.update(job=10, subspcorr=st_correlation,
+                      buflen=st_duration / info['sfreq'])
         logger.info('    Processing data using tSSS with st_duration=%s'
-                    % st_duration)
+                    % max_st['buflen'])
         st_when = 'before' if st_fixed else 'after'  # relative to movecomp
     else:
-        st_duration = min(raw_sss.times[-1], 10.)  # chunk size
+        st_duration = raw_sss.time_as_index(10., use_rounding=True)  # chunksz
```
Comment (wronk) [inline mne/preprocessing/maxwell.py]: Maybe just put this on the line above if you want the comment. Staying under 80 char is now making this unclear :p

**Comment by wronk** [comment]
LGTM

**Comment by Eric89GXL** [comment]
Thanks, pushed a commit with better comments.

**Comment by agramfort** [comment]
+1 for merge too.

@Eric89GXL shall we backport?

**Comment by Eric89GXL** [comment]
Yeah we probably should, I can do that tomorrow

**Comment by Eric89GXL** [comment]
Rebased squashed and push-forced, will push the commit to `master` if it passes, and also cherry-pick and push to `maint/0.12`

**Comment by Eric89GXL** [comment]
Merged by rebase and pushing to 0.12 now, thanks for the quick reviews

**Pull request closed without merging**